test/router.spec.js

```diff
@@ -450,6 +450,56 @@ describe('Restify Router', function () {
 
   });
 
+  describe('Nested routers via .add should process middlewares in order', function () {
+    it('Should process middlewares in order', function (done) {
+      var v1 = new Router();
+      var auth = new Router();
+      var register = new Router();
+
+      var first = function (req, res, next) {
+        req.test = [1];
```

Comment:
This middleware should first check if req.test exists and is an array. If it is, append to the existing list, otherwise initialize it to [1].

That way this test will fail for current implementation. It will pass once you remove the 2nd line I mentioned.